delivery/services/dds_service.py

```diff
@@ -0,0 +1,113 @@
+                    'data', 'put',
+                    '--source', delivery_order.delivery_source,
+                    '-p', delivery_order.delivery_project,
+                    '--silent',
+                    ]
+
+            log.debug("Running dds with cmd: {}".format(" ".join(cmd)))
+
+            execution = external_program_service.run(cmd)
+            delivery_order.delivery_status = DeliveryStatus.delivery_in_progress
+            delivery_order.mover_pid = execution.pid
+            session.commit()
+
+            execution_result = yield external_program_service.wait_for_execution(execution)
+
+            if execution_result.status_code == 0:
+                delivery_order.delivery_status = DeliveryStatus.delivery_successful
+                log.info(f"Successfully delivered: {delivery_order}")
+            else:
+                delivery_order.delivery_status = DeliveryStatus.delivery_failed
+                error_msg = f"Failed to deliver: {delivery_order}. DDS returned status code: {execution_result.status_code}"
+                log.error(error_msg)
+                raise RuntimeError(error_msg)
+
+        except Exception as e:
+            delivery_order.delivery_status = DeliveryStatus.delivery_failed
+            log.error(f"Failed to deliver: {delivery_order} because this exception was logged: {e}")
+            raise e
+        finally:
+            # Always commit the state change to the database
+            session.commit()
+
+    @gen.coroutine
+    def deliver_by_staging_id(self, staging_id, delivery_project, md5sum_file, token_path, skip_mover=False):
+
+        stage_order = self.staging_service.get_stage_order_by_id(staging_id)
+        if not stage_order or not stage_order.status == StagingStatus.staging_successful:
+            raise InvalidStatusException("Only deliver by staging_id if it has a successful status!"
+                                         "Staging order was: {}".format(stage_order))
+
+        delivery_order = self.delivery_repo.create_delivery_order(delivery_source=stage_order.get_staging_path(),
+                                                                  delivery_project=delivery_project,
+                                                                  delivery_status=DeliveryStatus.pending,
+                                                                  staging_order_id=staging_id,
+                                                                  md5sum_file=md5sum_file)
+
+        args_for_run_dds_put = {'delivery_order_id': delivery_order.id,
+                              'delivery_order_repo': self.delivery_repo,
+                              'external_program_service': self.mover_external_program_service,
+                              'session_factory': self.session_factory,
+                              'token_path': token_path,
+                              'dds_conf': self.dds_conf,
+                              }
+
+        if skip_mover:
```

Comment:
We should make a ticket about renaming this parameter to "skip_delivery" or something later.

Reply:
Is a Github issue OK or would you prefer to have everything in Jira?

Reply:
I would at least like a ticket in Jira, but both would be great. The Jira one could be a pointer to the github issue.